src/dialogs/create-file.tsx

```diff
@@ -0,0 +1,233 @@
+          className="file-create-modal"
+          onEscapePress={handleEscape}
+          footer={
+              <>
+                  <Button
+                    variant="primary"
+                    isDisabled={!filename || initial_text === "" || filename_error !== null}
+                    onClick={handleSave}
+                  >
+                      {_("Create")}
+                  </Button>
+                  <Button variant="link" onClick={() => dialogResult.resolve("")}>
+                      {_("Cancel")}
+                  </Button>
+              </>
+          }
+        >
+            <Stack>
+                {create_error &&
+                <Alert
+                  className="file-editor-alert"
+                  variant="danger"
+                  title={create_error}
+                  isInline
+                />}
+                <Form isHorizontal>
+                    <FormGroup
+                      label={_("File name")}
+                      fieldId="file-name"
+                    >
+                        <TextInput
+                          autoFocus // eslint-disable-line jsx-a11y/no-autofocus
+                          id="file-name"
+                          isRequired
+                          value={filename}
+                          onChange={(_event, value) => {
+                              setFileNameError(checkFilename(value, cwdInfo?.entries || {}, undefined));
+                              setFilename(value);
+                          }}
+                        />
+                        <FormHelper fieldId="file-name" helperTextInvalid={filename_error} />
+                    </FormGroup>
+                    <TextArea
+                      id="editor-text-area"
+                      className="file-editor"
+                      value={initial_text}
+                      onChange={(_ev, content) => setInitialText(content)}
+                    />
+                    {candidates.length > 0 &&
+                        <FormGroup fieldId="create-file-owner" label={_("Owner")}>
+                            <FormSelect
+                              id='create-file-owner'
+                              value={owner}
+                              onChange={(_ev, val) => setOwner(val)}
+                            >
+                                {candidates.map(owner =>
+                                    <FormSelectOption
+                                      key={owner}
+                                      value={owner}
+                                      label={owner}
+                                    />)}
+                            </FormSelect>
+                        </FormGroup>}
+                </Form>
+            </Stack>
+        </Modal>
+    );
+};
+
+export async function show_create_file_dialog(dialogs: Dialogs, path: string,
+    addAlert: (title: string, variant: AlertVariant, key: string, detail?: string) => void) {
```

Comment:
Could you align the attributes, and move `path` into its own line as well? Makes it easier to read, otherwise this `addAlert` looks like part of the code.